tests/test_commonMetrics.py

```diff
@@ -202,6 +202,27 @@ def testBadlyTypedKeys(self):
         with self.assertRaises(MetricComputationError):
             task.run(metadata)
 
+    def testOldMetadata(self):
+        """Test compatibility with version 0 metadata
+
+        This can't actually test differences in unit handling between version 0
+        and version 1, but at least verifies that the code didn't choke on
+        old-style metadata.
+        """
+        newMetadata = self.scienceTask.getFullMetadata()
+        oldMetadata = newMetadata.copy()
+        for key in newMetadata.names(topLevelOnly=False):
+            if "__version__" in key:
+                oldMetadata.remove(key)
+
+        result = self.task.run(oldMetadata)
+        lsst.pipe.base.testUtils.assertValidOutput(self.task, result)
+        meas = result.measurement
+
+        self.assertIsInstance(meas, Measurement)
+        self.assertEqual(meas.metric_name, self.metric)
+        self.assertGreater(meas.quantity, 0.0 * u.byte)
```

Comment:
Don't we know the lowest number we would get? With new metadata we would calculate actual bytes. With old metadata format we could get 1024 times larger, but the floor is the actual bytes from new metadata.

Reply:
With the timing/memory metrics I've found it dangerous to make assumptions -- the results are too dependent on the execution environment, and even occasional false failures in unit tests are a big problem in our workflow.

Given that, choosing a higher floor would _look_ more meaningful but would still need to be much lower than actual memory usage, meaning there's no real constraint.

Reply:
You have no way to analyze the same task metadata in the old way and the new way and do a direct comparison from the two approaches? The first will use system units and the second will always return bytes.